pkg/chain/ethereum/ethutil/resubscribe_test.go

```diff
@@ -0,0 +1,280 @@
+package ethutil
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/event"
+)
+
+func TestEmitOriginalError(t *testing.T) {
+	backoffMax := 100 * time.Millisecond
+	alertThreshold := 100 * time.Millisecond
+
+	failedOnce := false
+	resubscribeFn := func(ctx context.Context) (event.Subscription, error) {
+		if !failedOnce {
+			failedOnce = true
+			return nil, fmt.Errorf("wherever I go, he goes")
+		}
+		delegate := event.NewSubscription(func(unsubscribed <-chan struct{}) error {
+			return nil
+		})
+		return delegate, nil
+	}
+
+	thresholdViolated := make(chan time.Duration, 10)
+	subscriptionFailed := make(chan error, 10)
+	subscription := WithResubscription(
+		backoffMax,
+		resubscribeFn,
+		alertThreshold,
+		func(elapsed time.Duration) { thresholdViolated <- elapsed },
+		func(err error) { subscriptionFailed <- err },
+	)
+	<-subscription.Err()
+
+	// Subscription failed one time so there should be one error in the channel.
+	subscriptionFailCount := len(subscriptionFailed)
+	if subscriptionFailCount != 1 {
+		t.Fatalf(
```

Comment:
Can we replace `t.Fatalf` with `t.Errorf` in all added tests so the test reports all verification failures instead of aborting on the first one?

Reply:
The reason I used `Fatalf` in this case is that we should not execute the other check if we do not want to risk the test to hang. If this channel is empty, `err := <-subscriptionFailed` execution in the next line will hang. For other cases, I do agree we can use `Errorf`.

Reply:
See 4841e64672d7215b060ca21758d9576d44f292e5 please.